enterprise/app/jobs/captain/documents/crawl_job.rb

```diff
@@ -2,17 +2,63 @@ class Captain::Documents::CrawlJob < ApplicationJob
   queue_as :low
 
   def perform(document)
-    if InstallationConfig.find_by(name: 'CAPTAIN_FIRECRAWL_API_KEY')&.value.present?
+    if pdf_document?(document)
+      perform_pdf_extraction(document)
+    elsif InstallationConfig.find_by(name: 'CAPTAIN_FIRECRAWL_API_KEY')&.value.present?
       perform_firecrawl_crawl(document)
     else
       perform_simple_crawl(document)
     end
   end
 
+  def pdf_document?(document)
+    return false if document.nil?
+
+    pdf_by_metadata?(document) || pdf_by_url?(document)
+  end
+
   private
 
   include Captain::FirecrawlHelper
 
+  def pdf_by_metadata?(document)
+    pdf_by_source_type?(document) || pdf_by_content_type?(document)
+  rescue StandardError
+    false
+  end
```

Comment:
Why is a rescue being used here?Standard errors shouldn’t be caught. Let them fail and raise naturally or use ChatwootExceptionTracker instead.

If there’s a specific exception you need to handle, catch only that explicitly.